### PR TITLE
Enable simulcast by default, add jest tests

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: src/room/RTCEngine.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export default class RTCEngine extends EventEmitter {
   close() {
     this.isClosed = true;
 
-    if (this.publisher) {
+    if (this.publisher && this.publisher.pc.signalingState !== 'closed') {
```
Comment (noahlt) [inline src/room/RTCEngine.ts:90]: Style: could use optional chaining (`?.`) here

File: src/room/track/options.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export interface TrackPublishDefaults {
   dtx?: boolean;
 
   /**
-   * use simulcast, defaults to false.
+   * use simulcast, defaults to true.
    * When using simulcast, LiveKit will publish up to three version of the stream
```
Comment (noahlt) [inline src/room/track/options.ts:31]: typo: "up to three version**s**"

File: src/room/defaults.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ import {
 
 let publishDefaults: TrackPublishDefaults = {
   audioBitrate: AudioPresets.speech.maxBitrate,
+  dtx: true,
```
Comment (noahlt) [inline src/room/defaults.ts:7]: defaulting `dtx` to `true` here appears to be a drive-by fix (because the [docs already say](https://github.com/livekit/client-sdk-js/blob/ac95ada/src/room/track/options.ts#L25) it defaults to true)

File: src/room/participant/LocalParticipant.ts [hunk 4]
```diff
@@ -470,147 +470,151 @@ export default class LocalParticipant extends Participant {
...
+
+export const presetsScreenShare = [
+  ScreenSharePresets.vga,
+  ScreenSharePresets.hd_8,
+  ScreenSharePresets.hd_15,
+  ScreenSharePresets.fhd_15,
+  ScreenSharePresets.fhd_30,
+];
+
+export function computeVideoEncodings(
```
Comment (noahlt) [inline src/room/participant/LocalParticipant.ts:499]: ❓ this is no longer a private member method; I assume this change was necessary to make it testable?

(I'm a little surprised Jest doesn't have a faculty for testing private methods, but then again it's a newfangled feature)

File: src/room/track/options.ts (hunk 2 above)
Comment (noahlt) [reply src/room/track/options.ts:31]: It may be worth noting that this flag only has an effect if both `width` and `height` are set.

File: src/room/participant/LocalParticipant.ts [hunk 5]
```diff
@@ -470,147 +470,151 @@ export default class LocalParticipant extends Participant {
...
-            maxBitrate: videoEncoding.maxBitrate,
-            /* @ts-ignore */
-            maxFramerate: videoEncoding.maxFramerate,
-          },
-        ];
-      }
+    // if resolution is high enough, we would send [q, h, f] res..
+    // otherwise only send [q, h]
+    // NOTE:
+    //   1. Ordering of these encodings is important. Chrome seems
```
Comment (noahlt) [inline src/room/participant/LocalParticipant.ts:536]: 😂

File: src/room/participant/LocalParticipant.ts [hunk 6]
```diff
@@ -470,147 +470,151 @@ export default class LocalParticipant extends Participant {
...
-      const preset = presets[i];
-      if (width >= preset.width && height >= preset.height) {
-        encoding = preset.encoding;
-      }
-    }
+  return encodings;
+}
 
-    return encoding;
+export function determineAppropriateEncoding(
```
Comment (noahlt) [inline src/room/participant/LocalParticipant.ts:591]: ❓ Did you export this (and `presetsForResolution`, below) with the intention of testing them, as well?

Also, I'm not sure how our doc generation works, but if these are exported will they automatically show up in our public docs?

File: src/room/participant/LocalParticipant.ts [hunk 7]
```diff
@@ -470,147 +470,151 @@ export default class LocalParticipant extends Participant {
...
 
-    if ((!videoEncoding && !useSimulcast) || !width || !height) {
-      // don't set encoding when we are not simulcasting and user isn't restricting
-      // encoding parameters
-      return undefined;
-    }
+  if (!videoEncoding) {
+    // find the right encoding based on width/height
+    videoEncoding = determineAppropriateEncoding(isScreenShare, width, height);
+    log.debug('using video encoding', videoEncoding);
```
Comment (noahlt) [inline src/room/participant/LocalParticipant.ts:522]: ❗ stray debug log

File: src/room/participant/LocalParticipant.test.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { VideoPresets } from '../..';
+import { computeVideoEncodings } from './LocalParticipant';
+
+describe('computeVideoEncodings', () => {
+  it('returns nil non-simulcast', () => {
+    const encodings = computeVideoEncodings(false, 640, 480, {
+      simulcast: false,
+    });
+    expect(encodings).toBeUndefined();
```
Comment (noahlt) [inline src/room/participant/LocalParticipant.test.ts:9]: ❓ What are the implications when `computeVideoEncodings` returns `undefined` here?  More specifically, I imagine if we return `undefined` here that some other part of the system (perhaps the browser) picks an encoding for us?

**Review by noahlt — COMMENTED** [review]
Some questions and minor comments

File: src/room/RTCEngine.ts (hunk 1 above)
Comment (davidzhao) [reply src/room/RTCEngine.ts:90]: are you thinking `this.publisher?.pc.signalingState !== 'closed'`? 

I thought about that but it wouldn't do what I want here: publisher.pc is defined *and* not closed.

File: src/room/track/options.ts (hunk 2 above)
Comment (davidzhao) [reply src/room/track/options.ts:31]: width and height are internally provided, where as this is options exposed to end users.

File: src/room/defaults.ts (hunk 3 above)
Comment (davidzhao) [reply src/room/defaults.ts:7]: yup.. it's already defaulting to true, just making it explicit for those that are reading this code.

File: src/room/participant/LocalParticipant.ts (hunk 4 above)
Comment (davidzhao) [reply src/room/participant/LocalParticipant.ts:499]: right, doing this for testing.. I've read something about using rewire to accomplish this.. but it seemed too magical and I wanted to avoid introducing another dev dependency.

File: src/room/participant/LocalParticipant.ts (hunk 6 above)
Comment (davidzhao) [reply src/room/participant/LocalParticipant.ts:591]: yup, coming up in a different PR, but great call on docs! I forgot about that. will mark them as internal.

File: src/room/participant/LocalParticipant.test.ts (hunk 8 above)
Comment (davidzhao) [reply src/room/participant/LocalParticipant.test.ts:9]: the browser picks its defaults when encodings aren't defined.

File: src/room/RTCEngine.ts (hunk 1 above)
Comment (noahlt) [reply src/room/RTCEngine.ts:90]: Ah, right, because with optional chaining when `this.publisher` is undefined it ends up comparing `undefined !== 'closed'`.
